### PR TITLE
Enabling application insights events

### PR DESCRIPTION
**Pull request by timharrison-moj**
Amendments to logback-spring.xml, properties and run script to enable application insight events to be generated.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -77,8 +77,8 @@ dependencies {
     compile("org.springframework.security:spring-security-jwt:1.0.10.RELEASE")
 
     compile("net.logstash.logback:logstash-logback-encoder:5.2")
-    implementation 'com.microsoft.azure:applicationinsights-logging-logback:2.4.0-BETA'
-    implementation 'com.microsoft.azure:applicationinsights-spring-boot-starter:1.2.0-BETA'
+    compile("com.microsoft.azure:applicationinsights-logging-logback:2.4.0-BETA")
```
Comment (petergphillips) [inline build.gradle]: latest gradle syntax is to use `implementation`, not `compile`.  Don't need the brackets since it is groovy and single quotes preferred when you don't need to do interpolation - i.e. revert this change :grin:

File: README.md [hunk 2]
```diff
@@ -224,12 +224,26 @@ Environment:
...
              -d -t mojdigitalstudio/community-proxy:latest`
 
 [ For PRODUCTION - supply all values - no defaults ]
 
 
+# Run Locally (outside of Docker)
+
+Supply the information required (detailed in the run.sh, which docker uses)
+
+java -DAPPLICATION_INSIGHTS_IKEY=xxxxx \
```
Comment (petergphillips) [inline README.md]: wouldn't specify the app insights key when running locally otherwise you will pollute t2 / t3 environments unnecessarily

File: run.sh [hunk 3]
```diff
@@ -1,12 +1,13 @@
 #!/bin/bash
 
-JAVA_OPTS="-Djavax.net.ssl.trustStore=/app/trusted.jks -Djavax.net.ssl.trustStorePassword=${TRUST_STORE_PASSWORD} -Djavax.net.ssl.trustStoreType=jks"
+JAVA_OPTS="-Djavax.net.ssl.trustStore=/app/trusted.jks \
+           -Djavax.net.ssl.trustStorePassword=${TRUST_STORE_PASSWORD} \
+           -Djavax.net.ssl.trustStoreType=jks \
+           -Dcom.sun.management.jmxremote.local.only=false \
+           -Ddelius.endpoint.url=${DELIUS_ENDPOINT_URL} \
+           -Ddelius.api.username=${DELIUS_API_USERNAME} \
+           -DAPPLICATION_INSIGHTS_IKEY=${APP_INSIGHTS_KEY} \
```
Comment (petergphillips) [inline run.sh]: I think this is confusing.  None of the other apps specify the insights key in their `run.sh`. Since the environment variable will be set in the docker container anyway there is no need to pass it in here

File: run.sh [hunk 4]
```diff
@@ -1,12 +1,13 @@
 #!/bin/bash
 
-JAVA_OPTS="-Djavax.net.ssl.trustStore=/app/trusted.jks -Djavax.net.ssl.trustStorePassword=${TRUST_STORE_PASSWORD} -Djavax.net.ssl.trustStoreType=jks"
+JAVA_OPTS="-Djavax.net.ssl.trustStore=/app/trusted.jks \
+           -Djavax.net.ssl.trustStorePassword=${TRUST_STORE_PASSWORD} \
+           -Djavax.net.ssl.trustStoreType=jks \
+           -Dcom.sun.management.jmxremote.local.only=false \
+           -Ddelius.endpoint.url=${DELIUS_ENDPOINT_URL} \
```
Comment (petergphillips) [inline run.sh]: don't think these two properties are needed, since they are set as environment variables instead

File: run.sh [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
+           -Ddelius.api.username=${DELIUS_API_USERNAME} \
+           -DAPPLICATION_INSIGHTS_IKEY=${APP_INSIGHTS_KEY} \
+           -Djwt.public.key=${JWT_PUBLIC_KEY} \
```
Comment (petergphillips) [inline run.sh]: spring will automatically convert `JWT_PUBLIC_KEY` into `jwt.public.key` anyway so not needed if set in the environment

File: src/main/resources/application.yml [hunk 6]
```diff
@@ -43,4 +45,4 @@ delius:
     username: DefaultValue
 
 azure.application-insights.web.enable-W3C: true
-azure.application-insights.web.enable-W3C-backcompat-mode: true
\ No newline at end of file
+azure.application-insights.web.enable-W3C-backcompat-mode: true
```
Comment (petergphillips) [inline src/main/resources/application.yml]: better to define as yml i.e.:```
azure:
  application-insights:
    web:
      enable-W3C: true```, don't think you need backwards compatibility?

File: src/main/resources/logback-spring.xml [hunk 7]
```diff
@@ -39,8 +39,21 @@
...
+        <appender-ref ref="consoleAppender"/>
+        <appender-ref ref="aiAppender"/>
+    </logger>
+
     <root level="INFO">
         <appender-ref ref="consoleAppender"/>
         <appender-ref ref="aiAppender"/>
     </root>
+
+    <springProfile name="ai">
```
Comment (petergphillips) [inline src/main/resources/logback-spring.xml]: Why is this necessary?  We don't have a spring profile of `ai` do we?

File: src/main/resources/logback-spring.xml [hunk 8]
```diff
@@ -39,8 +39,21 @@
         <appender-ref ref="aiAppender"/>
     </logger>
 
+    <logger name="com.microsoft.applicationinsights" additivity="false" level="INFO">
```
Comment (petergphillips) [inline src/main/resources/logback-spring.xml]: I take it this was just for debugging what was going wrong at the time - is it still needed or is it best to decrease to `WARN`?

File: build.gradle (hunk 1 above)
Comment (timharrison-moj) [reply build.gradle]: I was eliminating differences from other services, but its not this, so happy to re-instate.

File: README.md (hunk 2 above)
Comment (timharrison-moj) [reply README.md]: ok. I'll move it out of the example but put a note in that this is how you would do it to test that events are being received. Thanks.

File: run.sh (hunk 4 above)
Comment (timharrison-moj) [reply run.sh]: Do you mean the DELIUS_* vars?

File: run.sh (hunk 3 above)
Comment (timharrison-moj) [reply run.sh]: I think its explicitly declaring what values are being passed from the environment to the JVM which is arguably one more line than necessary but adds clarity for a simple soul like me.

File: run.sh (hunk 4 above)
Comment (timharrison-moj) [reply run.sh]: As above, explicitly passing values from environment to JVM.

File: run.sh (hunk 5 above)
Comment (timharrison-moj) [reply run.sh]: Same thing here - explicit definition of values passed from environment.  I will canvass a coupe of other opinions to see which people feel most comfortable with, thanks.

File: src/main/resources/application.yml (hunk 6 above)
Comment (timharrison-moj) [reply src/main/resources/application.yml]: Its a change Simon just put in - happy to remove if its not required?

Comment (petergphillips) [reply src/main/resources/application.yml]: sorry, I saw the line had been modified so thought you had changed it, please ignore

File: src/main/resources/logback-spring.xml (hunk 7 above)
Comment (timharrison-moj) [reply src/main/resources/logback-spring.xml]: We don't. And its not. Another part of eliminating the differences between this and elite2 around the app insights config. Elite2 does have this section though, though have not used or seen an "ai" spring profile there either.  I've removed it.

File: src/main/resources/logback-spring.xml (hunk 8 above)
Comment (timharrison-moj) [reply src/main/resources/logback-spring.xml]: It was part of eliminating differences - now removed and checked that events are still generated. The core issue was in the way the key was being set.

File: run.sh (hunk 3 above)
Comment (petergphillips) [reply run.sh]: would be better to therefore call it `APPLICATION_INSIGHTS_IKEY` instead of `APP_INSIGHTS_KEY`, since when it is converted into an automatic deployment it will be the only app with a different environment variable and I presume if someone creates the environment with `APPLICATION_INSIGHTS_IKEY` it won't won't work since it will be overridden in `run.sh` :disappointed:

**Review by petergphillips — APPROVED** [review]
:+1:
